internal/juju/applications.go

```diff
@@ -1226,6 +1227,24 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 		}
 	}
 
+	if input.UnsetConfig != nil {
+		// these are config entries to be unset
+		if len(input.UnsetConfig) > 0 {
```

Comment:
todo: Only `len(input.UnsetConfig) > 0` is needed. Remove the other for less nesting.